arrow/src/compute/kernels/take.rs

```diff
@@ -876,6 +900,48 @@ mod tests {
         .unwrap();
     }
 
+    #[test]
+    fn test_take_primitive_nullable_indices_non_null_values_with_offset() {
+        let index =
+            UInt32Array::from(vec![Some(0), Some(1), Some(2), Some(3), None, None]);
+        let index = index.slice(2, 4);
+        let index = index.as_any().downcast_ref::<UInt32Array>().unwrap();
+
+        assert_eq!(
+            index,
+            &UInt32Array::from(vec![Some(2), Some(3), None, None])
+        );
+
+        test_take_primitive_arrays_non_null::<Int64Type>(
+            vec![0, 1, 2, 3, 4, 5, 6],
```

Comment:
Would it make sense to use different values here than the indices -- perhaps something like

```suggestion
            vec![0, 10, 20, 30, 40, 50, 60],
```

So it is clearer from just this context that just `20` and `30` should be returned

Reply:
Good idea, done